angular/openmrs-header/openmrs-header.component.js

```diff
@@ -7,22 +7,81 @@
  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
  * graphic logo is a trademark of OpenMRS Inc.
  */
+
+import openmrsRest from './../openmrs-rest/openmrs-rest.js';
+import ngDropdwons from 'angular-dropdowns'
+import css from './openmrs-header.css'
+
 var template = require('./openmrs-header.html');
 var openmrs = require('../../src/scss/images/openmrs-with-title-small.png');
 
-export default angular.module('openmrs-contrib-uicommons.header', []).component('openmrsHeader', {
+export default angular.module('openmrs-contrib-uicommons.header', ['openmrs-contrib-uicommons.rest', 'ngDropdowns']).component('openmrsHeader', {
   template: template,
   controller: OpenmrsHeaderController,
   controllerAs: 'vm',
   bindings: {
-    title : '<'
+
   }
 }).name;
 
-OpenmrsHeaderController.$inject=[];
+OpenmrsHeaderController.$inject=['openmrsRest'];
 
-function OpenmrsHeaderController() {
+function OpenmrsHeaderController(openmrsRest) {
 	var vm = this;
+
+    vm.locationList = [];
+    vm.selectedLocation = {};
+    vm.disableLocationDropdown = false;
+    vm.sessionContext;
+
+    vm.accountOptions = [
+        {
+            option: 'My Account',
+            href: '/openmrs/adminui/myaccount/myAccount.page'
+        }
+    ];
+    vm.selectedOption = {};
+
+
+
+    vm.activate = activate;
+    vm.changeLocation = changeLocation;
+    vm.isLocationListEmpty = isLocationListEmpty;
+
+    activate();
+
+    function activate() {
+        openmrsRest.listFull('location').then(function (response) {
+            angular.forEach(response.results, function(location){
+                angular.forEach(location.tags, function(tag){
+                    if(tag.display === "Login Location"){
+                        vm.locationList.push(location);
+                    }
+                });
+            });
+            vm.isLocationListEmpty();
+        });
+
+        openmrsRest.get('appui/session').then(function(response){
+            vm.sessionContext = response;
```

Comment:
I think we'll need to expose a new REST resource in webservices-rest, which returns running modules to check for presence of the reference application module. Please post to talk in case there's another/better way to support that.